src/wallet/wallet2.cpp

```diff
@@ -8521,42 +8522,47 @@ static lns_prepared_args prepare_tx_extra_loki_name_system_values(wallet2 const
     }
 
     boost::optional<std::string> failed;
-    std::vector<cryptonote::COMMAND_RPC_LNS_NAMES_TO_OWNERS::response_entry> response = wallet.lns_names_to_owners(request, failed);
+    if (response) {
+      *response = wallet.lns_names_to_owners(request, failed);
+    } else {
+      std::vector<cryptonote::COMMAND_RPC_LNS_NAMES_TO_OWNERS::response_entry> response_vec = wallet.lns_names_to_owners(request, failed);
+      response = &response_vec;
```

Comment:
This takes a dangling ptr, after this inner scope closes the contents is trashed. I usually handle this by doing instead at the top of the function.

```cpp
std::vector<cryptonote::COMMAND_RPC_LNS_NAMES_TO_OWNERS::response_entry> response_;
if (!response)
{
    response = &response_;
}

// ... rest of the function uses response->
```

That replaces the pointer at the start of the function if needed and guarantees you won't deal with nullptrs, then the rest of the function can continue using response as if it was never null. On scope exit everything gets deleted and is opaque to the caller.

Reply:
Thank you!!

Updated Scope